File: 0901_1000/0926/0926.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.readline())
    s = [f.readline().strip() for i in range(n)]
    k = sum(sum(l == 'C' for l in line) for line in s) / 2
    a = []
    for line in s:
        for l in line:
            a.append('1' if k else '2')
            if k and l == 'C':
                k -= 1
        a.append('\n')
    q.write(''.join(a))

File: 0901_1000/0926/0926.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, k = 0;
	char *s;

	fscanf(f, "%d", &n);
	s = (char*)malloc(n * n + 1);
	for (int i = 0; i < n; ++i)
		fscanf(f, "%s", s + i * n);
	for (int i = 0; i < n * n; ++i)
		k += (s[i] == 'C');
	k = k / 2 + 1;
	for (int i = 0; i < n; ++i, putc('\n', q))
		for (int j = 0; j < n; ++j, putc(k ? '1' : '2', q))
			if (k && s[i * n + j] == 'C')
				--k;
	return 0;
}
